### PR TITLE
Update: use control-plane instead of master for node selector

### PR DESCRIPTION
**Pull request by zewolfe**
Towards - https://github.com/giantswarm/roadmap/issues/2135

### CODE REVIEW

**Review by bdehri — APPROVED** [review]

File: helm/coredns-app/templates/deployment-masters.yaml [hunk 1]
```diff
@@ -55,8 +55,13 @@ spec:
               topologyKey: kubernetes.io/hostname
       nodeSelector:
         {{-  range $key, $value := $.Values.mastersInstance.nodeSelector }}
-        {{ $key }}: {{ $value }}
-        {{- end}}
+          {{- if and (eq $key "node-role.kubernetes.io/control-plane") (lt (int $.Capabilities.KubeVersion.Minor) 25) }}
```
Comment (webwurst) [inline helm/coredns-app/templates/deployment-masters.yaml:58]: Maybe we can use `(semver "1.2.3").Compare` here? https://docs.helm.sh/docs/chart_template_guide/function_list/#semver

Comment (webwurst) [inline helm/coredns-app/templates/deployment-masters.yaml:58]:
```suggestion
          {{- if and (eq $key "node-role.kubernetes.io/control-plane") (semverCompare "<1.25" .Capabilities.KubeVersion.Version) }}
```

**Comment by webwurst** [comment]
I see the linter complaining, not sure why. I took the `semverCompare` from [ingress-nginx/templates/controller-service.yaml](https://github.com/kubernetes/ingress-nginx/blob/main/charts/ingress-nginx/templates/controller-service.yaml#L69):
```
{{- if and (semverCompare ">=1.20" .Capabilities.KubeVersion.Version) (.Values.controller.service.appProtocol) }}
```

**Comment by zewolfe** [comment]
I think it was the scope, because `.Capabilities` isn't available in the block. I've updated it to reference from `$.Capabilities` instead. I've updated it and it looks okay.

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**